net/dns/resolvd.go

```diff
@@ -0,0 +1,53 @@
+// Copyright (c) 2020 Tailscale Inc & AUTHORS All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+//go:build openbsd
+// +build openbsd
+
+package dns
+
+import (
+	"os/exec"
+
+	"tailscale.com/types/logger"
+)
+
+func newResolvdManager(logf logger.Logf, interfaceName string) (*resolvdManager, error) {
+	return &resolvdManager{
+		logf:   logf,
+		ifName: interfaceName,
+	}, nil
+}
+
+// resolvdManager is an OSConfigurator which uses route(1) to teach OpenBSD's
+// resolvd(8) about DNS servers.
+type resolvdManager struct {
+	logf   logger.Logf
+	ifName string
+}
+
+func (m *resolvdManager) SetDNS(config OSConfig) error {
+	args := []string{
+		"nameserver",
+		m.ifName,
+	}
+
+	for _, s := range config.Nameservers {
+		args = append(args, s.String())
+	}
+
+	cmd := exec.Command("/sbin/route", args...)
+	return cmd.Run()
+}
+func (m *resolvdManager) SupportsSplitDNS() bool {
+	return true
```

Comment:
I don't think this is correct, based on the implementation of SetDNS. Split DNS support here means that the overall manager can tell the OS to only use a set of resolvers for specific domain suffixes. The implementation of SetDNS doesn't make use of MatchDomains, so it wouldn't implement the configuration that the DNS manager wanted.

Can resolvd do per-suffix DNS routes? If so, this can be true, but SetDNS needs to obey MatchDomains in what it configures.

Reply:
I don't think resolvd can do per-suffix DNS routes. I am not completely sure I understand DNSR, but it seems that resolved's approach of ordering nameservers based off route priority _almost_ achieves the same thing (but maybe that's just my lack of understanding :P).

I guess making this `false` is the correct path forward.